### PR TITLE
fixed l1_normalize showing disprancy with jax and numpy

### PR DESCRIPTION
**Pull request by akshatvishu**
The current version of `l1_normalize` is showing discrepancy between `numpy` , `jax` v/s `torch`, `tf` backends.

```python
import jax
import jax.numpy as jnp
import paddle
import torch
import tensorflow as tf
import ivy
import ivy.functional.backends.jax as jb
import ivy.functional.backends.numpy as nb 
import ivy.functional.backends.torch as tb 
import ivy.functional.backends.tensorflow as tfb 
import ivy.functional.backends.paddle as pb


a = nb.experimental.l1_normalize(np.array([[-1., -1.], [-1., -2.]], dtype=np.float32), axis=0)
print(f"numpy output: {a}")

b = tfb.experimental.l1_normalize(tf.constant([[-1., -1.], [-1., -2.]], dtype=tf.float32), axis=0)
print(f"tf output: {b}")

c = ivy.experimental.l1_normalize(ivy.array([[-1., -1.], [-1., -2.]], dtype=ivy.float32), axis=0)
print(f"ivy output: {c}")

d = jb.experimental.l1_normalize(jax.numpy.array([[-1., -1.], [-1., -2.]], dtype=jax.numpy.float32), axis=0)
print(f"jax output: {d}")

e = tb.experimental.l1_normalize(torch.tensor([[-1., -1.], [-1., -2.]], dtype=torch.float32), axis=0)
print(f"torch output: {e}")

f = pb.experimental.l1_normalize(paddle.to_tensor([[-1., -1.], [-1., -2.]], dtype=paddle.float32), axis=0)
print(f"paddle output: {f}")

"""
numpy output: 
[[-0.5        -0.33333334]
 [-0.5        -1.3333334 ]]

tf output: 
[[-0.5        -0.33333334]
 [-0.5        -0.6666667 ]]

ivy output: 
ivy.array([[-0.5       , -0.33333334],
       [-0.5       , -1.33333337]])

jax output: 
[[-0.5        -0.33333334]
 [-0.5        -1.3333334 ]]

torch output: 
tensor([[-0.5000, -0.3333],
        [-0.5000, -0.6667]])

paddle output: Tensor(shape=[2, 2], dtype=float32, place=Place(cpu), stop_gradient=True,
       [[-0.50000000, -0.33333334],
        [-0.50000000, -0.66666669]])
"""
```

### CODE REVIEW

**Comment by akshatvishu** [comment]
Hey @MahmoudAshraf97   I don't why this PR added label `paddle backend` because I think this is more related to other `ivy backend` in-general ! I didn't even touch `paddle` backend implementation because it was working fine

File: ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_norms.py
Comment (MahmoudAshraf97) [inline ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_norms.py:1]: Why are these changes necessary?

File: ivy/functional/backends/paddle/experimental/norms.py
Comment (MahmoudAshraf97) [inline ivy/functional/backends/paddle/experimental/norms.py:1]: why is float16 not supported? you can use functions from ivy paddle backend to mitigate this

File: ivy/functional/backends/jax/experimental/norms.py [hunk 1]
```diff
@@ -11,11 +11,10 @@ def l1_normalize(
...
 ) -> JaxArray:
     if axis is None:
-        norm = jnp.sum(jnp.abs(x.flatten()))
-        denorm = norm * jnp.ones_like(x)
+        denorm = jnp.linalg.norm(x.flatten(), 1, axis)
     else:
-        norm = jnp.sum(jnp.abs(x), axis=axis, keepdims=True)
-        denorm = jnp.divide(norm, jnp.abs(x) + 1e-12)
+        denorm = jnp.linalg.norm(x, 1, axis, keepdims=True)
+        denorm = jnp.maximum(denorm, 1e-12)
```
Comment (MahmoudAshraf97) [inline ivy/functional/backends/jax/experimental/norms.py:17]: this is not needed since you can divide by 0

File: ivy/functional/backends/numpy/experimental/norms.py [hunk 2]
```diff
@@ -13,12 +13,11 @@ def l1_normalize(
...
     if axis is None:
-        norm = np.sum(np.abs(x.flatten()))
-        denorm = norm * np.ones_like(x)
+        denorm = np.linalg.norm(x.flatten(), 1, axis)
     else:
-        norm = np.sum(np.abs(x), axis=axis, keepdims=True)
-        denorm = np.divide(norm, np.abs(x) + 1e-12)
-    return np.divide(x, denorm)
+        denorm = np.linalg.norm(x, 1, axis, keepdims=True)
+        denorm = np.maximum(denorm, 1e-12)
```
Comment (MahmoudAshraf97) [inline ivy/functional/backends/numpy/experimental/norms.py:19]: same as jax

**Review by MahmoudAshraf97 — CHANGES_REQUESTED** [review]

File: ivy/functional/backends/paddle/experimental/norms.py
Comment (akshatvishu) [reply ivy/functional/backends/paddle/experimental/norms.py:1]: Hey @MahmoudAshraf97  I am very sorry for replying so late(I was on a vacation ) . The `float16` is not supported by [paddle.linalg.norm](https://www.paddlepaddle.org.cn/documentation/docs/en/api/paddle/linalg/norm_en.html#norm) hence, I added it to `unsupported_dtypes`

File: ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_norms.py
Comment (akshatvishu) [reply ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_norms.py:1]: Tests were failing with the updated changes hence, I need to tweak the test to make it work!

File: ivy/functional/backends/paddle/experimental/norms.py
Comment (MahmoudAshraf97) [reply ivy/functional/backends/paddle/experimental/norms.py:1]: You should fix the failure then not change the test 😁

File: ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_norms.py
Comment (MahmoudAshraf97) [reply ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_norms.py:1]: You should fix the failure then not change the test 😁
Mainly you need to test for all dtypes not just float

**Review by MahmoudAshraf97 — CHANGES_REQUESTED** [review]

File: ivy/functional/backends/tensorflow/experimental/norms.py [hunk 3]
```diff
@@ -1,9 +1,26 @@
...
+                "float64",
+                "int8",
+                "int16",
+                "int32",
+                "int64",
+            )
+        }
+    },
+    backend_version,
+)
```
[the remark this answers is not in the document]
Comment (akshatvishu) [reply ivy/functional/backends/tensorflow/experimental/norms.py]: I  passed the wrong supported `dypes` ; as [tf.norm](https://www.tensorflow.org/api_docs/python/tf/norm) only accepts ` float32, float64, complex64, complex128`. Thus, I  still  kept the `with_supported_device_and_dtypes`

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

File: ivy/functional/backends/jax/experimental/norms.py [hunk 4]
```diff
@@ -1,21 +1,30 @@
...
     axis: Optional[int] = None,
     out: Optional[JaxArray] = None,
 ) -> JaxArray:
-    if axis is None:
-        norm = jnp.sum(jnp.abs(x.flatten()))
-        denorm = norm * jnp.ones_like(x)
-    else:
-        norm = jnp.sum(jnp.abs(x), axis=axis, keepdims=True)
-        denorm = jnp.divide(norm, jnp.abs(x) + 1e-12)
+    denorm = jnp.linalg.norm(x, 1, axis, keepdims=True)
```
Comment (MahmoudAshraf97) [inline ivy/functional/backends/jax/experimental/norms.py:27]: You need to handle the case when `axis=None` as this function accepts None axes only when the array is less than 3D

**Review by MahmoudAshraf97 — CHANGES_REQUESTED** [review]

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.
